sdk/storage/Azure.Storage.Files.Shares/src/ShareClientOptions.cs

```diff
@@ -124,6 +124,23 @@ public enum ServiceVersion
         /// </summary>
         public TransferValidationOptions TransferValidation { get; } = new();
 
+        /// <summary>
+        /// If set to true, trailing dot (.) will be allowed to suffex directory and file names.
+        /// If false, the trailing dot will be trimmed.
+        /// Supported by x-ms-version 2022-11-02 and above.
+        /// </summary>
+        public bool? AllowTrailingDot { get; set; }
+
+        /// <summary>
+        /// If set to true, trailing dot (.) will be allowed to source file names.
+        /// If false, the trailing dot will be trimmed.
+        /// Supported by x-ms-version 2022-11-02 and above.
+        /// Applicable to <see cref="ShareFileClient.Rename(string, Models.ShareFileRenameOptions, System.Threading.CancellationToken)"/>,
+        /// <see cref="ShareFileClient.UploadRangeFromUriAsync(System.Uri, HttpRange, HttpRange, System.Threading.CancellationToken)"/>,
+        /// and <see cref="ShareFileClient.StartCopyAsync(System.Uri, Models.ShareFileCopyOptions, System.Threading.CancellationToken)"/>.
+        /// </summary>
+        public bool? SourceAllowTrailingDot { get; set; }
```

Comment:
TODO update comment, this parameter also applies to `Directory Rename`

Reply:
Consider moving this parameter to the options bags for the few APIs that actually need it.

Reply:
Fixed.